telematic_system/telematic_historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/service_manager.py

```diff
@@ -116,26 +119,65 @@ async def get_file_path_from_nats(self, msg):
         data = msg.data.decode()
         msg_json_object = json.loads(data)
 
-        rosbag_name = msg_json_object["filename"]
+        rosbag_path = msg_json_object["filepath"]
 
-        rosbag_path = Path(self.config.upload_destination_path) / rosbag_name
         # Add rosbag name to queue
         self.rosbag_queue.append(rosbag_path)
 
-    async def process_rosbag(self):
+    async def process_rosbag_queue(self):
         # This task is responsible for processing the rosbag in the queue - As long as the queue is not empty - keep processing
         while True:
             if not self.rosbag_parser.is_processing and self.rosbag_queue:
-                self.rosbag_parser.process_rosbag(self.update_first_rosbag_status())
+                # Get the mysql entry name: org-name/
+                rosbag_complete_path = Path(self.rosbag_queue[0])
+                rosbag_mysql_entry = str(rosbag_complete_path.relative_to(Path(self.config.upload_destination_path)))
+
+                #Update mysql status to Processing
+                self.update_mysql_entry(rosbag_mysql_entry, ProcessingStatus.IN_PROGRESS.value)
+
+                processing_status, processing_err_msg = self.rosbag_parser.process_rosbag(self.update_first_rosbag_status())
+                # Update mysql entry for rosbag based on whether processing was successful or not
+                self.update_mysql_entry(rosbag_mysql_entry, processing_status, processing_err_msg)
 
-                # TODO: Update mysql entry for rosbag based on whether processing was successful
             await asyncio.sleep(1.0)
 
+    def create_mysql_conn(self):
+
+        try:
+            conn = mysql.connector.connect(user= self.config.mysql_user, password= self.config.mysql_password,
+                              host= self.config.mysql_host,
+                              database= self.config.mysql_db, port = self.config.mysql_port)
+            self.config.logger.info("Connected to MySQL database!")
+            return conn
+        except mysql.connector.Error as err:
+            if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
+                self.config.logger.error("Mysql User name or password not accepted")
```

Comment:
Add the value of user in the error message? The same for the rest of the exceptions?